src/exo/API.py

```diff
@@ -341,7 +341,7 @@ def c_code_str(self):
         return decls + "\n" + defns
 
     def compile_c(self, directory: Path, filename: str):
-        compile_procs([self._loopir_proc], directory, f"{filename}.c", f"{filename}.h")
+        compile_procs([self], directory, f"{filename}.c", f"{filename}.h")
```

Comment:
I remember seeing this and we did talk about it but I forgot; was this a bug?

Reply:
It was untested code; someone changed the interface of compile_procs at some point and as a result this call was incorrect.